### DISCUSSION TITLE
Same device appearing multiple times in a single BT Scan result

### DISCUSSION THREAD

**Discussion by munkii** [shinyorg/shiny#1454] [Q&A] [answered]
During our move to 3.x we chnaged how we connect to BT devices and now we always scan for a Device first. This is how I had implemented the Scan 

```
ScanConfig scanConfig = this.GetScanConfig(deviceType);
IDeviceWrapper deviceWrapper = null;
this.InternalStartScan(
                        scanConfig,
                        results =>
                        {
                            System.Diagnostics.Debug.WriteLine("IsScanning: " + this.IsScanning);
                            System.Diagnostics.Debug.WriteLine("Adapter Wrapper ScanResult");
                            if (results.Count == 1)
                            {
                                ScanResult scanResult = results[0];

                                if (scanResult.Peripheral.Uuid != null &&
                                            string.Compare(scanResult.Peripheral.Uuid, uuid.ToString(), true) == 0)
                                {
                                    this.StopScan();
                                    deviceWrapper = new DeviceWrapper(scanResult.Peripheral);

                                    this.DeviceFound?.Invoke(this, deviceWrapper);
                                }
                            }
                        },
                        (exception) =>
                        {
                            this.DeviceNotFound?.Invoke(this, EventArgs.Empty);
                        });
    }
```

Note that I checked the count of the results and only inspect it if it was 1.

There is a new model of device that is causing multiple entries to be returned in the scan or for the same device and the numbers vary all within the same scanning "task". Ranging from 30 to 20, then 10. 7, 2 0 and back up to 2 or 6 then finally after 50 or more seconds it would be just 1.

I have now changed my code and it inspects any set of results greater than 0 and looks for one instance of the unique device and connects to it.

Why does the same device get reported back multiple times in a single result and then with varying counts in subsequent result sets?

**Comment by aritchie** [comment, marked as answer]
So scan results will contain unique instances of the peripheral, but will get broadcast for every time that peripheral emits a signal.  The scan result will contain an updated RSSI and possibly any updated advertising data.

**Reply by aritchie** [reply]
You can use the managed scanner OR use a .Distinct(x => x.Peripheral.Uuid) (off the top of my head, but close to what you'll want)

**Reply by munkii** [reply]
I'm happy with just lookng at the ScanResult for the unique device id. Thanks for explaining why multiple entries are in the scan results.
